cpp/bindings/py_geometric_path.cpp

```diff
@@ -0,0 +1,46 @@
+#include <bindings.hpp>
+
+namespace toppra {
+namespace python {
+PyPiecewisePolyPath::PyPiecewisePolyPath(const Matrices& coefficients,
+                                         std::vector<value_type> breakpoints)
+    : m_path{coefficients, breakpoints} {};
+
+PyPiecewisePolyPath::PyPiecewisePolyPath(const PiecewisePolyPath& path)
+    : m_path(path){};
+
+Vector PyPiecewisePolyPath::eval_single(value_type x, int order) const {
+  return m_path.eval_single(x, order);
+};
+
+nparr PyPiecewisePolyPath::eval(const Vector& xs, int order) const {
+  auto ret = m_path.eval(xs, order);
```

Comment:
Did you try to let pybind11 convert the vector of Eigen matrices by itself ?

Given https://pybind11.readthedocs.io/en/stable/advanced/cast/stl.html and https://pybind11.readthedocs.io/en/stable/advanced/cast/eigen.html, I think it should be able to do it.
If you did this to have the same type as in the Python implementation, I understand.

Reply:
Pybind will convert Vector to np.ndarray, but the output type here is Vectors, which is converted to List[np.ndarray]

Reply:
For your info, you do not have to write a wrapper to achieve the same behavior. For function of `PiecewisePolyPath` that needs rework to be exposed in Python, you can write static function
```cpp
retType bind_function (PiecewisePolyPath& p, args...) {
res =  p.function (args);
// Do something with res and return it.
}
```
Then when binding
```cpp
py::class_<PiecewisePolyPath>(m, "PiecewisePolyPath")
.def ("dof", &PiecewisePolyPath::dof)
.def("function", bind_function)
;
```

You will have a lot less code. You can even write lambda functions (https://pybind11.readthedocs.io/en/stable/classes.html?highlight=lambda#binding-lambda-functions).